adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -807,8 +807,9 @@ protected void onCommandReceived(final ProtonSender sender, final CommandContext
 
         LOG.debug("creating command response [status: {}, correlation-id: {}, reply-to: {}]",
                 statusCode, correlationId, context.getResourceIdentifier().toString());
-        final CommandResponse commandResponse = CommandResponse.from(context.getMessagePayload(),
-                context.getMessageContentType(), statusCode, correlationId, context.getResourceIdentifier());
+        final CommandResponse commandResponse = CommandResponse.from(context.getTenantId(), context.getDeviceId(),
```

Comment:
I am afraid that this will not work if the device is not authenticated and the reply-to address specified by the Command indeed contains the device ID. However, the C&C API does not strictly require this but instead allows the usage of any arbitrary string after the `command/$tenant-id` ...

Reply:
@sophokles73 Thank you. How about on arrival of the commands, check if the device is not authenticated, extract the reply-id from the reply-to parameter and compare with the device-id. If it doesn't match then append `:~device-id~:` to the reply-to parameter. On arrival of the command response, check for for the pattern `:~ ~:` in reply-id. Then extract the device id, remove the pattern and send the response. WDYT?

Reply:
> How about on arrival of the commands, check if the device is not authenticated, extract the reply-id 

IMHO we should do this for *every* command and not rely on the fact that the device will also be authenticated when it sends the response ...

Apart from that, I think that your approach can work. However, I would prefer to use a similar approach as in the HTTP adapter, i.e. include the device ID at the position in the address that we usually expect it but e.g. prepend it with a single character (`0`/`1`) indicating whether the device ID needs to be removed from the address before forwarding the response downstream ...

Reply:
Good to know more about the approach in Http Adapter. I also thinks its better to have an uniform approach. I will go ahead with it.